`test_rechunk` and `test_unify_chunks` fail with `dask`/`dask-expr` 2024.8.2

**Describe the issue**:

[This exact dask change](https://github.com/dask/dask/pull/11337) is causing two `dask-expr` test to fail:
```
=========================== short test summary info ============================
FAILED dask_expr/array/tests/test_array.py::test_rechunk - TypeError: 'NoneType' object is not iterable
FAILED dask_expr/array/tests/test_array.py::test_unify_chunks - TypeError: 'NoneType' object is not iterable
========================= 2 failed, 31 passed in 1.74s =========================
```

Full logs (both tests fail the same way):

<details>

```
_________________________________ test_rechunk _________________________________

    def test_rechunk():
        a = np.random.random((10, 10))
        b = da.from_array(a, chunks=(4, 4))
        c = b.rechunk()
        assert c.npartitions == 1
>       assert_eq(b, c)

dask_expr/array/tests/test_array.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/6ap5gqscb55qifzqbgg54vizfb1xzap8-python3.12-dask-2024.8.2/lib/python3.12/site-packages/dask/array/utils.py:324: in assert_eq
    b, bdt, b_meta, b_computed = _get_dt_meta_computed(
/nix/store/6ap5gqscb55qifzqbgg54vizfb1xzap8-python3.12-dask-2024.8.2/lib/python3.12/site-packages/dask/array/utils.py:271: in _get_dt_meta_computed
    _check_dsk(x.dask)
dask_expr/_core.py:517: in dask
    return self.__dask_graph__()
dask_expr/array/core.py:54: in __dask_graph__
    return super().__dask_graph__()
dask_expr/_core.py:513: in __dask_graph__
    return toolz.merge(layers)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

dicts = [None, {('fromarray-5a753850e5ed9ac9082c521a46654e6c', 0, 0): array([[0.66733136, 0.79475484, 0.17277945, 0.56357861],...     [0.26039384, 0.5122637 , 0.64383969, 0.10626229],
       [0.9476847 , 0.94537969, 0.33568735, 0.65006998]]), ...}]
kwargs = {}, factory = <class 'dict'>, rv = {}, d = None

    def merge(*dicts, **kwargs):
        """ Merge a collection of dictionaries
    
        >>> merge({1: 'one'}, {2: 'two'})
        {1: 'one', 2: 'two'}
    
        Later dictionaries have precedence
    
        >>> merge({1: 2, 3: 4}, {3: 3, 4: 4})
        {1: 2, 3: 3, 4: 4}
    
        See Also:
            merge_with
        """
        if len(dicts) == 1 and not isinstance(dicts[0], Mapping):
            dicts = dicts[0]
        factory = _get_factory(merge, kwargs)
    
        rv = factory()
        for d in dicts:
>           rv.update(d)
E           TypeError: 'NoneType' object is not iterable

/nix/store/7knjbgfi9l8ksqn4n4p18iv0sdj48176-python3.12-toolz-0.12.1/lib/python3.12/site-packages/toolz/dicttoolz.py:39: TypeError
```

</details>

**Minimal Complete Verifiable Example**:

```sh
$ pytest
```

**Anything else we need to know?**:

I am tackling the `dask`(`-expr`) update on nixpkgs: https://github.com/NixOS/nixpkgs/pull/338759

**Environment**:

- Dask version: 2024.8.2
- Python version: 3.12
- Operating System: Linux (NixOS)
- Install method (conda, pip, source): nix ;p

--- a/dask_expr/array/rechunk.py
+++ b/dask_expr/array/rechunk.py
@@ -87,6 +87,9 @@ class Rechunk(Array):
 
     def _layer(self):
         method = self.method or dask.config.get("array.rechunk.method")
+        # Default to tasks since P2P is not implemented
+        if method is None:
+            method = "tasks"
         if method == "tasks":
             steps = plan_rechunk(
                 self.array.chunks,

